e2store/src/e2store/stream.rs

```diff
@@ -0,0 +1,89 @@
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::PathBuf,
+};
+
+use super::types::{Entry, Header};
+
+/// e2s.rs was built to load full .era/.era2 files into memory and provide a simple API to access
+/// the data. The issue for this is for larger files this wouldn't be feasible, as the entire file
+/// would need to be loaded into memory. This is where e2store_file.rs comes in, it provides a way
+/// to read and write e2store files in a streaming fashion.
+pub struct E2StoreStream {
+    pub e2store_file: File,
+}
+
+impl E2StoreStream {
+    pub fn new_read(e2store_path: &PathBuf) -> anyhow::Result<Self> {
```

Comment:
`new_read` and `new_write` methods are not intuitive to me and I prefer to stick with `open` and `create` which aligns with the standard Rust naming convection for opening files. Moving `File::create` inside `create` makes the API more clear IMO.

```rust
impl E2StoreStream {
    pub fn open(e2store_path: &PathBuf) -> Result<Self> {
        let e2store_file = File::open(e2store_path)?;
        Ok(Self { e2store_file })
    }

    pub fn create(e2store_path: &PathBuf) -> Result<Self> {
        let e2store_file = File::create(e2store_path)?;
        Ok(Self { e2store_file })
    }
    ...
}
```